src/ServiceManagerBase.sol

```diff
@@ -57,6 +62,32 @@ abstract contract ServiceManagerBase is IServiceManager, OwnableUpgradeable {
         _avsDirectory.updateAVSMetadataURI(_metadataURI);
     }
 
+    /**
+     * @notice Creates a new range payment on behalf of an AVS, to be split amongst the
+     * set of stakers delegated to operators who are registered to the `avs`.
+     * Note that the owner calling this function must have approved the tokens to be transferred to the ServiceManager
+     * and of course has the required balances.
+     * @param rangePayments The range payments being created
+     * @dev Expected to be called by the ServiceManager of the AVS on behalf of which the payment is being made
+     * @dev The duration of the `rangePayment` cannot exceed `paymentCoordinator.MAX_PAYMENT_DURATION()`
+     * @dev The tokens are sent to the `PaymentCoordinator` contract
+     * @dev Strategies must be in ascending order of addresses to check for duplicates
+     * @dev This function will revert if the `rangePayment` is malformed,
+     * e.g. if the `strategies` and `weights` arrays are of non-equal lengths
+     */
+    function submitRangePayments(
```

Comment:
curious, is there a reason we are not calling this function payForRange?
Do we have a policy of reusing same name function only when proxying the exact same call, whereas here we are also transferring tokens and approving?

Reply:
That was my original thought to rename it since it has some additional ERC20 token logic but its essentially proxying the same call. I renamed to `payForRange` for consistency sake with the AVSDirectory calls having same naming.

https://github.com/Layr-Labs/eigenlayer-middleware/pull/242/commits/cbe2107dd590fd49c5aa97274ca7beecb9b19644